operator/internal/handlers/lokistack_create_or_update.go

```diff
@@ -85,11 +86,20 @@ func CreateOrUpdateLokiStack(
 
 	timeoutConfig, err := manifests.NewTimeoutConfig(stack.Spec.Limits)
 	if err != nil {
-		ll.Error(err, "failed to parse query timeout")
-		return "", &status.DegradedError{
-			Message: fmt.Sprintf("Error parsing query timeout: %s", err),
-			Reason:  lokiv1.ReasonQueryTimeoutInvalid,
-			Requeue: false,
+		if errors.Is(err, manifests.ErrInvalidPerTenantConfig) {
+			ll.Error(err, "invalid per-tenant config")
+			return "", &status.DegradedError{
+				Message: fmt.Sprintf("Invalid per-tenant limits config: %s", err),
+				Reason:  lokiv1.ReasonInvalidPerTenantLimitsConfig,
+				Requeue: false,
+			}
+		} else {
+			ll.Error(err, "failed to parse query timeout")
+			return "", &status.DegradedError{
+				Message: fmt.Sprintf("Error parsing query timeout: %s", err),
+				Reason:  lokiv1.ReasonQueryTimeoutInvalid,
+				Requeue: false,
+			}
```

Comment:
No need to put it under a else
```suggestion
		}
		ll.Error(err, "failed to parse query timeout")
		return "", &status.DegradedError{
			Message: fmt.Sprintf("Error parsing query timeout: %s", err),
			Reason:  lokiv1.ReasonQueryTimeoutInvalid,
			Requeue: false,
```